src/workspaces/components/Tools.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import {useSelector} from "react-redux";
+import {getTools} from "../selectors";
+import Button from "@material-ui/core/Button";
+
+type ToolProps = { workspaceUri: string }
+
+const Tools: React.FC<ToolProps> = (props: ToolProps) => {
+    const tools = useSelector(getTools);
+    const handleClick = (e : React.MouseEvent) => { e.stopPropagation() };
+    return <>
+        {tools.map( t => <Button
```

Comment:
I believe there is a `key` property missing, React should throw a warning in this case